apps/server/src/modules/tool/external-tool/service/external-tool-logo-service.spec.ts

```diff
@@ -0,0 +1,310 @@
+					const { externalTool } = setup();
+
+					const func = () => service.validateLogoSize(externalTool);
+
+					expect(func).not.toThrow();
+				});
+			});
+		});
+
+		describe('when external tool has no given logo', () => {
+			const setup = () => {
+				const externalTool: ExternalTool = externalToolFactory.build();
+
+				return { externalTool };
+			};
+
+			it('should not throw an error', () => {
+				const { externalTool } = setup();
+
+				const func = () => service.validateLogoSize(externalTool);
+
+				expect(func).not.toThrow();
+			});
+		});
+	});
+
+	describe('fetchLogo', () => {
+		describe('when tool has no logo url', () => {
+			const setup = () => {
+				const externalTool: ExternalTool = externalToolFactory.buildWithId({ logoUrl: undefined });
+
+				return {
+					externalTool,
+				};
+			};
+
+			it('should not fetch the logo', async () => {
+				const { externalTool } = setup();
+
+				const logo = await service.fetchLogo(externalTool);
+
+				expect(logo).toBeUndefined();
+			});
+		});
+
+		describe('when tool has a logo url', () => {
+			const setup = () => {
+				const externalTool: ExternalTool = externalToolFactory.buildWithId();
+				const logoBuffer: Buffer = Buffer.from('logo content', 'utf-8');
+				const base64Logo = logoBuffer.toString('base64');
+
+				httpService.get.mockReturnValue(
+					of({
+						data: logoBuffer,
+						status: HttpStatus.OK,
+						statusText: 'OK',
+						headers: {},
+						config: {},
+					} as AxiosResponse<ArrayBuffer>)
+				);
+
+				const logoUrl = 'https://logo.com/';
+
+				return {
+					externalTool,
+					base64Logo,
+					logoUrl,
+				};
+			};
+
+			it('should fetch logo', async () => {
+				const { externalTool, logoUrl } = setup();
+
+				await service.fetchLogo(externalTool);
+
+				expect(httpService.get).toHaveBeenCalledWith(logoUrl, { responseType: 'arraybuffer' });
+			});
+
+			it('should log the fetched url', async () => {
+				const { externalTool, logoUrl } = setup();
+
+				await service.fetchLogo(externalTool);
+
+				expect(logger.info).toHaveBeenCalledWith(new ExternalToolLogoFetchedLoggable(logoUrl));
+			});
+
+			it('should return base64 encoded logo', async () => {
+				const { externalTool, base64Logo } = setup();
+
+				const logo: string | undefined = await service.fetchLogo(externalTool);
+
+				expect(logo).toBe(base64Logo);
+			});
+		});
+
+		describe('when error occurs on fetching logo', () => {
+			const setup = () => {
+				const externalTool: ExternalTool = externalToolFactory.buildWithId();
+
+				httpService.get.mockReturnValue(
+					throwError(() => new HttpException('Failed to fetch logo', HttpStatus.NOT_FOUND))
+				);
+
+				return {
+					externalTool,
+				};
+			};
+
+			it('should throw error', async () => {
+				const { externalTool } = setup();
+
+				const func = () => service.fetchLogo(externalTool);
+
+				await expect(func()).rejects.toThrow(ExternalToolLogoFetchFailedLoggableException);
```

Comment:
Please also check the code from your mocked exception here

Reply:
i am mocking the httpException and test if the loggable is thrown... what did is miss here?

Reply:
The status code of the exception from httpService. That should be the same in both exceptions